client/cli/src/libra_client.rs

```diff
@@ -276,32 +179,30 @@ impl LibraClient {
         }
     }
 
-    // Retrieves and
-    // - If `with_state_change`, will also retrieve state proof from node and update trusted_state accordingly
+    /// Retrieves account state
+    /// - If `with_state_proof`, will also retrieve state proof from node and update trusted_state accordingly
     pub fn get_account_state(
         &mut self,
         account: AccountAddress,
         with_state_proof: bool,
     ) -> Result<(Option<AccountView>, Version)> {
-        let method = "get_account_state".to_string();
-        let params = vec![serde_json::json!(account)];
+        let client_version = self.trusted_state.latest_version();
+        let mut batch = JsonRpcBatch::new();
+        batch.add_get_account_state_request(account);
+        if with_state_proof {
+            batch.add_get_state_proof_request(client_version);
+        }
+        let responses = self.client.execute(batch)?;
 
-        let response = if with_state_proof {
-            self.get_with_state_proof(method, serde_json::json!(params))
-        } else {
-            self.client.send_libra_request(method, params)
-        };
+        if with_state_proof {
+            let state_proof = get_response_from_batch(1, &responses)?.as_ref();
+            self.process_state_proof_response(state_proof)?;
+        }
 
-        match response {
+        match get_response_from_batch(0, &responses)? {
             Ok(result) => {
-                let account_view = match result {
-                    serde_json::Value::Null => None,
-                    _ => {
-                        let account_view: AccountView = serde_json::from_value(result)?;
-                        Some(account_view)
-                    }
-                };
-                Ok((account_view, self.trusted_state.latest_version()))
+                let account_view = AccountView::optional_from_response(result.clone())?;
+                Ok((account_view, client_version))
```

Comment:
the `client_version` returned here is before `process_state_proof_response` which is stale